Content.Client/Doors/DoorSystem.cs

```diff
@@ -138,14 +139,17 @@ private void UpdateAppearanceForDoorState(Entity<DoorComponent> entity, SpriteCo
         }
     }
 
-    private void UpdateSpriteLayers(SpriteComponent sprite, string baseRsi)
+    private void UpdateSpriteLayers(SpriteComponent old, string prototype)
     {
-        if (!_resourceCache.TryGetResource<RSIResource>(SpriteSpecifierSerializer.TextureRoot / baseRsi, out var res))
-        {
-            Log.Error("Unable to load RSI '{0}'. Trace:\n{1}", baseRsi, Environment.StackTrace);
+        if (!_prototypeManager.TryIndex(prototype, out var proto))
+            return;
+
+        if (!proto.TryGetComponent(out SpriteComponent? sprite, _componentFactory))
             return;
-        }
 
-        sprite.BaseRSI = res.RSI;
+        foreach (var layer in old.AllLayers)
+        {
+            layer.Rsi = sprite.BaseRSI;
+        }
```

Comment:
once again wondering why we need to overwrite the RSI of every layer individually instead of updating the base RSI of the sprite component